### PR TITLE
Fix #1

### LINKED ISSUE #1: Implement command line parser
Implement a command line parser with following options:
+ `-d <directory>` or `--directory <directory>` : the directory to serve (default to current directory)
+ `-p <port>` or `--port <port>` : the port to serve on (default to 70)
+ `-h` or `--hidden` : show hidden files if specified

### PR DESCRIPTION
**Pull request by skprabhanjan**
@Hrily , Changed it accordingly to fix #1 but need to install boost library wherever this is used.
Here are a few links to the same :)
https://www.boost.org/doc/libs/1_50_0/doc/html/quickbook/install.html
https://www.boost.org/doc/libs/1_55_0/more/getting_started/unix-variants.html
https://www.boost.org/doc/libs/1_62_0/more/getting_started/windows.html

Please review this and let me know if anything needs to be changed.
PS : please check the latest commit with message ("Fixes #1 - typo fixed")
Thanks :)

### CODE REVIEW

File: server.cpp [hunk 1]
```diff
@@ -305,26 +325,65 @@ void FileGopherServer::start(int n_connections){
...
 // Driver function.
-int main(int argc, char *argv[]){
+int main(int argc, char *argv[])
+{
+
     // Print Usage.
-    if(argc != 2){
-        printf("USAGE: \n\tserver <directory-to-serve>\n\n");
-        printf("Where <directory-to-serve> root directory of server.\n");
+    if (argc != 3)
```
Comment (Hrily) [inline server.cpp]: I think you can remove this if statement since all the arguments are optional.

File: server.cpp [hunk 2]
```diff
@@ -268,33 +285,36 @@ FileGopherServer::FileGopherServer(string root, int port) {
...
 }
 
 /**
  * Function to start the server.
  * @param n_connections The number of simultaneos connections.
  *                      Default 5.
  */
-void FileGopherServer::start(int n_connections){
-    if(listen(socket_fd, n_connections))
+void FileGopherServer::start(int n_connections)
```
Comment (Hrily) [inline server.cpp:298]: Can you print an info statement here of following format:
`Serving <directory> at 0.0.0.0:<port>`

File: server.cpp [hunk 3]
```diff
@@ -305,26 +325,65 @@ void FileGopherServer::start(int n_connections){
...
+
+    // variables needed to be captured from command line options
+
+    string directory = "."; // defaults to cwd
+    int port = 70;          // default port value of 70
+    bool show_hidden_files = false;
+
+    // Declare the supported options.
+    po::options_description desc("Allowed options");
+    desc.add_options()("directory,d", po::value<string>(), "the directory to serve")("port,p", po::value<int>(), "the port to serve on (default : 70)")("hidden,h", po::value<bool>(), "show hidden files if specified");
```
Comment (Hrily) [inline server.cpp]: Follow 80 char/line limit

File: server.cpp (hunk 1 above)
Comment (skprabhanjan) [reply server.cpp]: Done

File: server.cpp (hunk 2 above)
Comment (skprabhanjan) [reply server.cpp:298]: Done

File: server.cpp (hunk 3 above)
Comment (skprabhanjan) [reply server.cpp]: Done

**Comment by skprabhanjan** [comment]
@Hrily , Done with all changes :)

File: server.cpp (hunk 3 above)
Comment (Hrily) [reply server.cpp]: Is this done?

Can't see the change

Comment (skprabhanjan) [reply server.cpp]: Sorry Editor just applied the wordwrap but was not actually a new line, fixed this now:)

File: server.cpp [hunk 4]
```diff
@@ -305,26 +326,57 @@ void FileGopherServer::start(int n_connections){
...
+{
+    // variables needed to be captured from command line options
+
+    string directory = "."; // defaults to cwd
+    int port = 70;          // default port value of 70
+    bool show_hidden_files = false;
+
+    // Declare the supported options.
+    po::options_description desc("Allowed options");
+    desc.add_options()("directory,d", po::value<string>(), "the directory to
```
Comment (Hrily) [inline server.cpp:345]: Did you compile this and check?

I think the non-terminated string at the end should throw an error.

**Comment by skprabhanjan** [comment]
Will close this and create a new PR.
The two PR's caused some confusion.

**Pull request closed without merging**